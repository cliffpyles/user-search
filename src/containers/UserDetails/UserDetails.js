import React, { Component } from 'react'
import { connect } from 'react-redux'
import { executeGetDetails } from '../../store/actions'
import Page from '../../components/Page'
import Card from '../../components/Card'

class UserDetails extends Component {
  componentDidMount() {
    if (this.props && this.props.executeGetDetails) {
      const { username, executeGetDetails } = this.props
      executeGetDetails(username)
    }
  }

  render() {
    const { followers, login } = this.props

    return (
      <Page title="User Details">
        <Card>
          <p>login: {login}</p>
        </Card>
      </Page>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...state.details,
    ...ownProps
  }
}

const mapDispatchToProps = {
  executeGetDetails
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserDetails)
