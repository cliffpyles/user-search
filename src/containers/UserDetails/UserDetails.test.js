import React from 'react'
import Renderer from 'react-test-renderer'
import initializeStore from '../../store'
import UserDetails from './'

test('can render user details', () => {
  const store = initializeStore({
    details: {
      login: 'nraboy'
    }
  })
  const result = Renderer.create(<UserDetails store={store} />)

  expect(result.toJSON()).toMatchSnapshot()
})
