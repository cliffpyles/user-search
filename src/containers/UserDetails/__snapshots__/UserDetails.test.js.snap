// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render user details 1`] = `
<div
  className="Page"
>
  <h1>
    User Details
  </h1>
  <div
    className="Card"
  >
    <p>
      login: 
      nraboy
    </p>
  </div>
</div>
`;
